### PR TITLE
chore: remove maintenance service interface and use machine service

### PR DESCRIPTION
**Pull request by Unix4ever**
Now maintenance service implements `MachineService` interface, stubbing
all not implemented methods.

Fixes: https://github.com/talos-systems/talos/issues/2764

Signed-off-by: Artem Chernyshev <artem.0xD2@gmail.com>

### CODE REVIEW

File: internal/app/machined/internal/server/v1alpha1/v1alpha1_server.go [hunk 1]
```diff
@@ -67,6 +69,7 @@ const OSPathSeparator = string(os.PathSeparator)
 
 // Server implements the gRPC service server.
 type Server struct {
+	storaged.Server
```
Comment (andrewrynhard) [inline internal/app/machined/internal/server/v1alpha1/v1alpha1_server.go:72]: I think we can leave this out from here for now. I plan on following up with `storaged` service.

Comment (Unix4ever) [reply internal/app/machined/internal/server/v1alpha1/v1alpha1_server.go:72]: ok, so should I leave machine service without this method for now?

Comment (Unix4ever) [reply internal/app/machined/internal/server/v1alpha1/v1alpha1_server.go:72]: I would like to share this logic between storage service and maintenance service later on though.

Comment (andrewrynhard) [reply internal/app/machined/internal/server/v1alpha1/v1alpha1_server.go:72]: yeah, I can make it common to them. It is just that the storage service will be in `storaged` and not in `machined` so we might as well skip in `machined` for now.

**Review by andrewrynhard — APPROVED** [review]

**Comment by andrewrynhard** [comment]
Not a blocker for me, but a note for the future. I think this would better be marked as a `refactor`. A chore is designated for things like CI changes. Non-code changes.

**Comment by Unix4ever** [comment]
ah, ok

**Pull request merged**